streamPair.q

```diff
@@ -0,0 +1,80 @@
+// load tables
+tab1: 40000#1_ flip `dateTime`bid`ask`bidVol`askVol!("*FFFF";",") 0: `:C:/q/dash/sample/data/stocks/USA500IDXUSD.csv;
+tab2: 40000#1_ flip `dateTime`bid`ask`bidVol`askVol!("*FFFF";",") 0: `:C:/q/dash/sample/data/stocks/USATECHIDXUSD.csv;
+//tab1: 1_ flip `dateTime`bid`ask`bidVol`askVol`Vol!("DFFFFF";",") 0: `:C:/q/dash/sample/data/stocks/EWA2.csv;
+//tab2: 1_ flip `dateTime`bid`ask`bidVol`askVol`Vol!("DFFFFF";",") 0: `:C:/q/dash/sample/data/stocks/EWC2.csv;
+tab3: flip `dateTime`spread`mean`up`low`operation!("P"$();"F"$();"F"$();"F"$();"F"$();"F"$());
+
+
+// Fix Time
+update dateTime:"P"$@[;19;:;"."] each dateTime from `tab1;
+update dateTime:"P"$@[;19;:;"."] each dateTime from `tab2;
+
+// Take date and log(prices)
+priceX: distinct select distinct dateTime, log bid, log ask from tab1;
+priceY: distinct select distinct dateTime, log bid, log ask from tab2;
+spreads: 200#select from tab3;
+
+// Eliminate nulls and filter data
+update delta:0f^deltas dateTime from `priceX;
+update delta:0f^deltas dateTime from `priceY;
+priceX: 0!1_priceX;
+priceY: 0!1_priceY;

```

Comment:
Perhaps you could combine all these queries into just one query to make it more efficient

Reply:
ok I will